src/Controller/AuthController.php

```diff
@@ -2,24 +2,28 @@
 
 namespace eLife\Journal\Controller;
 
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
-use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Security\Http\Util\TargetPathTrait;
 
 final class AuthController extends Controller
 {
-    public function redirectAction() : Response
+    use TargetPathTrait;
+
+    public function redirectAction(Request $request) : Response
     {
         if (!$this->isGranted('FEATURE_CAN_AUTHENTICATE')) {
             throw new NotFoundHttpException('Not found');
         }
 
-        $request = $this->get('request_stack')->getCurrentRequest();
-        $path['_forwarded'] = $request->attributes;
-        $path['_controller'] = 'HWIOAuthBundle:Connect:redirectToService';
-        $path['service'] = 'elife';
-        $subRequest = $request->duplicate([], null, $path);
+        if ($referer = trim($request->headers->get('Referer'))) {
+            $firewall = $this->get('security.firewall.map')->getFirewallConfig($request);
+            $this->saveTargetPath($request->getSession(), $firewall->getName(), $referer);
```

Comment:
Is there any validation on the `Referer` that checks the host matches for example? Is it coincevable for someone external to send a user with its own `Referer` and getting it redirected there (and is it a problem?)

Reply:
Yep, already present https://github.com/elifesciences/journal/blob/f2ccd2f9b64e42494fbd8fb527cda5ed55beb91e/test/Controller/AuthenticationTest.php#L86-L119